### ISSUE TITLE
wc/require-listener-teardown false positive when using AbortController with addEventListener({ signal })

### ISSUE THREAD

**Issue by MRyves** [43081j/eslint-plugin-wc#148]
### Summary

The rule wc/require-listener-teardown reports an error for event listeners added in connectedCallback() even when they are torn down via AbortController. Using the signal option on addEventListener is a standards-based teardown mechanism; calling abort() removes the listener. This should be treated as compliant teardown, but it currently triggers the rule.

### ESLint / Plugin versions

- eslint-plugin-wc: 3.0.1 (also reproducible on 3.0.0)
- ESLint: (e.g.) 9.x (also seen on 8.x)
- Node: (e.g.) v20.x
- Environment: browser / Web Components (Lit 3).

### Rule
`wc/require-listener-teardown`

### Steps to reproduce
```
import {LitElement} from 'lit';

class XEl extends LitElement {
  private abortController = new AbortController();

  connectedCallback() {
    super.connectedCallback();
    this.addEventListener(
      'click',
      () => this.handleClick(),
      { signal: this.abortController.signal }
    );
  }

  disconnectedCallback() {
    super.disconnectedCallback();
    this.abortController.abort(); // removes the listener
  }

  private handleClick() {/* ... */}
}
customElements.define('x-el', XEl);
```

### Actual result:
wc/require-listener-teardown errors: “Event listeners attached in connectedCallback should be torn down during disconnectedCallback.”

### Expected result:
No error. The listener is torn down by calling abort() on the controller associated with the signal passed to addEventListener. This is equivalent to removing the listener manually
